### ISSUE TITLE
html: It does not work

### ISSUE THREAD

**Issue by ret80** [godotengine/godot#6147]
**Operating system or device - Godot version:**
godot 2.1

**Issue description** (what happened, and what was expected):
It looks like this:
![image](https://cloud.githubusercontent.com/assets/233268/17645349/057d803e-61ac-11e6-860d-7b754ff8917a.png)

but should like this:
![image](https://cloud.githubusercontent.com/assets/233268/17645336/d84e9cb0-61ab-11e6-9089-6041cf53ddab.png)

**Comment by eska014** [comment]
Is this 2D or 3D?
How does the player character work?
How does the background work?
Is the absence of the two lower buttons intentional or also a bug?
Are there other issues apart from those that I overlooked?

**Comment by bojidar-bg** [comment]
@eska014 Yes, that the screen is over 6 months old, so, might not apply to master :smiley: 

(TL;DT - Pretty bump, is it still valid in master?)

**Comment by ret80** [comment]
>1 Is this 2D or 3D?
2d and 3d (box - 3d, character - 2d, bkg - 2d)
> How does the player character work?
character - sprite3d
> How does the background work?
environment -> texture
> Is the absence of the two lower buttons intentional or also a bug?
You do not need to look at the buttons
> Are there other issues apart from those, that I overlooked?
no

**Comment by eska014** [comment]
Sprite3D uses `_draw_primitive` in the ES2 rasterizer, see here for my comment about that: https://github.com/godotengine/godot/issues/7176#issuecomment-276475083
The missing texture background is caused by our copy shader failing to compile on the web

**Comment by TazeTSchnitzel** [comment]
I also have this issue. It'd probably be best to retitle it “Background shader doesn't compile on web” or something.

**Comment by DoHe** [comment]
@leonkrause I'm experiencing the exact same issue on 3.1.2
